main.cpp

```diff
@@ -64,151 +63,592 @@ New/This/Pointers/References conclusion
+
+    int* value;
+
+    // primitives
+    IntType& add(int v);
+    IntType& subtract(int v);
+    IntType& multiply(int v);
+    IntType& divide(int v);
+    // FloatType args
+    IntType& add(const FloatType& ft);
+    IntType& subtract(const FloatType& ft);
+    IntType& multiply(const FloatType& ft);
+    IntType& divide(const FloatType& ft);
+    // DoubleType args
+    IntType& add(const DoubleType& dt);
+    IntType& subtract(const DoubleType& dt);
+    IntType& multiply(const DoubleType& dt);
+    IntType& divide(const DoubleType& dt);
+    // IntType args
+    IntType& add(const IntType& it);
+    IntType& subtract(const IntType& it);
+    IntType& multiply(const IntType& it);
+    IntType& divide(const IntType& it);
 };
 
-float FloatType::add(float lhs, float rhs)
+// FloatType member functions implementation
+// Primitives
+FloatType& FloatType::add(float v)
 {
-    return lhs + rhs;
+    std::cout << "(ft) " << *value << " + " << v << " = ";
+    *value += v;
+    std::cout << *value << '\n';
+    return *this;
 }
 
-float FloatType::subtract(float lhs, float rhs)
+FloatType& FloatType::subtract(float v)
 {
-    return lhs - rhs;
+    std::cout << "(ft) " << *value << " - " << v << " = ";
+    *value -= v;
+    std::cout << *value << '\n';
+    return *this;
 }
 
-float FloatType::multiply(float lhs, float rhs)
+FloatType& FloatType::multiply(float v)
 {
-    return lhs * rhs;
+    std::cout << "(ft) " << *value << " * " << v << " = ";
+    *value *= v;
+    std::cout << *value << '\n';
+    return *this;
 }
 
-float FloatType::divide(float lhs, float rhs)
+FloatType& FloatType::divide(float v)
 {
-    if (rhs == 0.0f)
+    if (v == 0.0f)
     {
         std::cout << "Error: cannot divide by 0!\n";
     }
-    return lhs / rhs;
+    std::cout << "(ft) " << *value << " / " << v << " = ";
+    *value /= v;
+    std::cout << *value << '\n';
+    return *this;
 }
 
-struct DoubleType
+// FloatType member functions implementation
+// FloatType argument
+FloatType& FloatType::add(const FloatType& ft)
 {
-    double add(double lhs, double rhs);
-    double subtract(double lhs, double rhs);
-    double multiply(double lhs, double rhs);
-    double divide(double lhs, double rhs);
-};
+    std::cout << "(ft) " << *value << " + (ft) " << *ft.value << " = ";
+    *value = *value + *ft.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+FloatType& FloatType::subtract(const FloatType& ft)
+{
+    std::cout << "(ft) " << *value << " - (ft) " << *ft.value << " = ";
+    *value = *value - *ft.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+FloatType& FloatType::multiply(const FloatType& ft)
+{
+    std::cout << "(ft) " << *value << " * (ft) " << *ft.value << " = ";
+    *value = *value * *ft.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+FloatType& FloatType::divide(const FloatType& ft)
+{
+    if (*ft.value == 0)
+    {
+        std::cout << "Error: cannot divide by 0!\n";
+    }
```

Comment:
floating point division-by-zero is well-defined, and returns `inf`.  It's not an error.  it's only an error for `int` division-by-zero

- [ ] correct warning